pkg/api/handlers/serverinstances.go

```diff
@@ -122,10 +136,49 @@ func (h *ServerInstancesHandler) DeleteServerInstance(req api.Context) error {
 
 func convertMCPServerInstance(instance v1.MCPServerInstance, serverURL string) types.MCPServerInstance {
 	return types.MCPServerInstance{
-		Metadata:     MetadataFrom(&instance),
-		UserID:       instance.Spec.UserID,
-		MCPServerID:  instance.Spec.MCPServerName,
-		MCPCatalogID: instance.Spec.MCPCatalogName,
-		ConnectURL:   fmt.Sprintf("%s/mcp-connect/%s", serverURL, instance.Name),
+		Metadata:                MetadataFrom(&instance),
+		UserID:                  instance.Spec.UserID,
+		MCPServerID:             instance.Spec.MCPServerName,
+		MCPCatalogID:            instance.Spec.MCPCatalogName,
+		MCPServerCatalogEntryID: instance.Spec.MCPServerCatalogEntryName,
+		ConnectURL:              fmt.Sprintf("%s/mcp-connect/%s", serverURL, instance.Name),
+	}
+}
+
+func (h *ServerInstancesHandler) AdminListServerInstancesForEntryInCatalog(req api.Context) error {
+	var instances v1.MCPServerInstanceList
+	if err := req.List(&instances, kclient.MatchingFields{
+		"spec.mcpServerCatalogEntryName": req.PathValue("entry_id"),
+		"spec.mcpCatalogName":            req.PathValue("catalog_id"),
+	}); err != nil {
+		return err
+	}
+
+	convertedInstances := make([]types.MCPServerInstance, 0, len(instances.Items))
+	for _, instance := range instances.Items {
+		convertedInstances = append(convertedInstances, convertMCPServerInstance(instance, h.serverURL))
+	}
```

Comment:
nit: might be worth it to extract this as a helper method that has a variadic argument for instances